cmd/ndm_daemonset/probe/changehandler.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"errors"
+
+	"github.com/openebs/node-disk-manager/blockdevice"
+	"k8s.io/klog"
+)
+
+func (pe *ProbeEvent) changeBlockDevice(bd *blockdevice.BlockDevice) error {
+	pe.Controller.FillBlockDeviceDetails(bd)
```

Comment:
@z0marlin You need to also apply the filter here. Otherwise the following can happen:
1. Detected a device for eg: /dev/loop1, which is ignored by the filter
2. User formatted the loop device and mounted it, 
3. Changehandler picks it up, fills the details, but while trying to update finds that the device resource does not exist and error out. This causes continuous erroring out and log flooding.

Reply:
is this happening because the cache stores all detected devices regardless of filters? the cache has loop devices too right?
One question though. I assumed UpdateBlockDevice will fail in case the bd is not found in etcd. What exactly is causing the log floods then?